### PR TITLE
lint: sloppyLen

### LINKED ISSUE #418: lint: useless len check
Name: `sloppyLen `
Before:
```go
len(arr) >= 0
len(arr) <= 0
len(arr) < 0
```
After:
```go
len(arr) > 0
len(arr) == 0
```

### PR DESCRIPTION
**Pull request by cristaloleg**
Fixes #418

### CODE REVIEW

File: lint/dummyLen_checker.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package lint
+
+//! Detects `regexp.Compile*` that can be replaced with `regexp.MustCompile*`.
+//
+// @Before:
+// re, _ := regexp.Compile(`const pattern`)
+//
+// @After:
+// re := regexp.MustCompile(`const pattern`)
```
Comment (Quasilyte) [inline lint/dummyLen_checker.go]: copy/paste detected. :)

File: lint/testdata/dummyLen/positive_tests.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package checker_test
+
+func f() {
+	a := []int{}
+
+	/// useless len comparison len(a) >= 0, always true
```
Comment (Quasilyte) [inline lint/testdata/dummyLen/positive_tests.go]: `useless len comparison` part is both redundant and somewhat offensive.
Maybe just `len(a) >= 0 is always true` is enough?
Same goes for other warnings.

File: lint/dummyLen_checker.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	"go/ast"
+	"go/token"
+
+	"github.com/go-toolsmith/astcopy"
+
+	"github.com/go-toolsmith/astfmt"
+)
+
+func init() {
+	addChecker(&dummyLenChecker{}, attrSyntaxOnly, attrExperimental)
```
Comment (Quasilyte) [inline lint/dummyLen_checker.go]: I don't like `dummy` name part.
It's too generic and doesn't seem to describe what checker does too well.
Quite hard to give right name from the first try, but maybe proper summary section can help us to find it.

**Comment by Quasilyte** [comment]
sloppyLen? :)

**Comment by Quasilyte** [comment]
I remember Russ Cox likes `sloppy` word. At least this is how I see it. :sweat_smile:

**Comment by cristaloleg** [comment]
Nice <3

**Comment by Quasilyte** [comment]
LGTM when checker is renamed.

**Pull request merged**